File: src/main/java/net/zenxarch/bot/defense/modules/Module.java
package net.zenxarch.bot.defense.modules;

import net.minecraft.client.MinecraftClient;
import net.minecraft.client.network.AbstractClientPlayerEntity;
import net.minecraft.entity.mob.MobEntity;
import net.minecraft.entity.projectile.ProjectileEntity;
import net.zenxarch.bot.defense.Settings;

public class Module {
  public static final MinecraftClient mc = MinecraftClient.getInstance();
  private String name;

  public Module(String name) {
    Settings.registerModule(name);
    Settings.registerBoolSetting(name, "enabled", true);
    this.name = name;
  }

  public void preTick(){};
  public void handleNone(){};
  public void handleProjectile(ProjectileEntity pe){};
  public void handleHostile(MobEntity me){};
  public void handlePlayer(AbstractClientPlayerEntity pe){};
  public void handlePassive(MobEntity me){};

  public boolean isActive() {
    return Settings.getBoolean(this.name + ".enabled");
  }

  public String getName() { return name; }
}

File: src/main/scala/net/zenxarch/bot/defense/modules/Module.scala
package net.zenxarch.bot.defense.modules

import net.minecraft.client.MinecraftClient
import net.minecraft.client.network.AbstractClientPlayerEntity
import net.minecraft.entity.mob.MobEntity
import net.minecraft.entity.projectile.ProjectileEntity
import net.zenxarch.bot.settings.Settings

abstract class Module(val name: String):

  Settings.registerSetting(name + ".enabled", true)

  def preTick() = {}
  def handleNone() = {}
  def handleProjectile(projectile: ProjectileEntity) = {}
  def handleHostile(mob: MobEntity) = {}
  def handlePlayer(player: AbstractClientPlayerEntity) = {}
  def handlePassive(mob: MobEntity) = {}

  def isActive() = Settings.getBoolean(this.name + ".enabled")

object Module:
  val mc = MinecraftClient.getInstance()
